rio_tiler/mosaic/reader.py

```diff
@@ -16,7 +17,7 @@ def mosaic_reader(
     assets: Sequence[str],
     reader: Callable,
     *args: Any,
-    pixel_selection: Optional[MosaicMethodBase] = None,
+    pixel_selection: Union[Type[MosaicMethodBase], MosaicMethodBase, None] = None,
```

Comment:
Can this be Optional[Union[Type[MosaicMethodBase], MosaicMethodBase]] ?

Reply:
Sure. Sometimes the nesting can get verbose